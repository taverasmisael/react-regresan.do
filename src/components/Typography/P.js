import React from 'react'
import PropTypes from 'prop-types'
import Typography from 'material-ui/Typography'

const P = ({ children, color = 'inherit', truncate = false, align = 'left' }) => (
  <Typography paragraph type="body2" color={color} align={align} noWrap={truncate}>
    {children}
  </Typography>
)

P.propTypes = {
  children: PropTypes.string,
  align: PropTypes.oneOf(['inherit', 'left', 'center', 'right', 'justify']),
  color: PropTypes.oneOf(['inherit', 'primary', 'secondary', 'accent', 'error', 'default']),
  truncate: PropTypes.bool
}

export default P
