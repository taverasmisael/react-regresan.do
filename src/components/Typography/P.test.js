import React from 'react'
import { shallow } from 'enzyme'

import P from './P'
const SuperLongText =
  'Lorem ipsum dolor sit amet consectetur adipisicing elit. Provident excepturi est quae distinctio dolorem. Nulla omnis inventore adipisci ipsam delectus quam voluptate, itaque neque, incidunt corporis nemo quasi, nisi sequi.'
describe('components/Typography/P', () => {
  describe('rendering', () => {
    it('renders without crashing', () => {
      shallow(<P />)
    })
    it('matches snapshot', () => {
      expect(<P />).toMatchSnapshot()
      expect(<P>Hello World</P>).toMatchSnapshot()
    })
  })

  describe('variants', () => {
    it('text display correctly', () => {
      const component = shallow(<P truncate color="accent">{SuperLongText}</P>)
      expect(component.props().color).toBe('accent')
    })
  })
})
