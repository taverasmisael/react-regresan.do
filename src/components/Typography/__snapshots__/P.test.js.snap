// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Typography/P rendering matches snapshot 1`] = `<P />`;

exports[`components/Typography/P rendering matches snapshot 2`] = `
<P>
  Hello World
</P>
`;
